src/jdk.incubator.foreign/share/classes/jdk/internal/foreign/Utils.java

```diff
@@ -85,4 +77,27 @@ public static VarHandle fixUpVarHandle(VarHandle handle) {
     private static MemoryAddressProxy filterAddress(MemoryAddress addr) {
         return (MemoryAddressImpl)addr;
     }
+
+    public static void checkRestrictedAccess() {
+        switch (foreignAccess) {
+            case "deny" -> throwIllegalAccessError(foreignAccess);
+            case "warn" -> System.err.println("WARNING: Accessing jdk.incubator.foreign.Foreign.");
+            case "debug" -> {
+                StringBuilder sb = new StringBuilder("DEBUG: Accessing jdk.incubator.foreign.Foreign.");
+                StackWalker.getInstance().walk(s -> {
+                    s
+                            .forEach(f -> sb.append(System.lineSeparator()).append("\tat ").append(f));
```

Comment:
Is this format intentional? Also it seems to me this can be replaced with

StackWalker().getInstance().forEach()?

Reply:
I didn't change this code, I just moved it around; I'll make another pass.